### DISCUSSION TITLE
Feedback: Sanity Docs Beta

### DISCUSSION THREAD

**Discussion by kmelve** [sanity-io/sanity#3330] [Sanity Studio Feedback]
We have shipped [a beta version of the new Sanity Docs](https://beta.sanity.io/docs). We are reworking the Sanity.io documentation for a couple of reasons:

* We want to improve the documentation information architecture, making it easier to use for new-comers and more efficient for users of the platform
* We needed to accommodate the documentation for the new Sanity Studio V3, which introduced new concepts and API surfaces
* We want the documentation to live in a separate app and not in the same as the marketing pages
* We want feedback on the ongoing documentation work to serve better people who are building on the platform

The documentation is still a work in progress, but we do want to hear from you:
* Can you migrate Studios from v2 to v3 based on the current documentation?
* What do you expect to be covered in the Sanity Studio docs?
* Is there anything that you think works well?

**Comment by surjithctly** [comment]
Need a detailed guide on Publish Actions in v3.

Eg: How to create an auto-generating slug based on the given fields using PublishAction. 

The v2 method is too complicated.

**Reply by surjithctly** [reply]
And thanks for finally enabling github discussions 🥳

**Reply by kmelve** [reply]
Unfortunately, we haven't done anything to the Document Actions API (yet). Only how you [import actions into the new configuration](https://beta.sanity.io/docs/platform/studio/v2-to-v3/document-actions-badges). But we appreciate the feedback!

**Comment by surjithctly** [comment]
Other areas to improve in docs:

1. Preview Pane changes in v3
2. Desk Structure in v3
3. Dashboard Config in v3
4. Detailed guide on every new v3 feature.
5. Theming Docs

**Reply by kmelve** [reply]
1. There aren't any changes to preview panes (or structure builder) outside of how you import structure to the new config, and the new `defaultDocumentNode` function that you can export from a config file.
2. The Structure API stays the same (see also 1)
3. The Dashboard Config API follows the plugin, which will be released soon! 
4. These are coming!
5. Themes API is also coming!

**Comment by plsrd** [comment]
Given that we can choose to use the Studio without Typescript, it would be nice to be able to toggle between a `.js` and `.ts` example in the docs. I can imagine non-Typescript community members having trouble with this.

**Reply by kmelve** [reply]
Yeah, this is a conundrum! I think we can make that happen, at least for the code block examples. For inline mention of files we could make some kind of inline object, but I wonder if that would be over-engineering it contra writing `sanity.config.(js|ts)` or just opt for being JavaScript-first and assume that TypeScript users know how to deal with it.

**Reply by plsrd** [reply]
That's a good point about the inline items! I like the idea of using `sanity.config.(js|ts)`. I'm a bit hesitant about opting for JS-first, since TypeScript compatibility is one of the the most exciting things about v3! We should show it off!

**Reply by matt416** [reply]
Perhaps a global toggle between ts and js examples (similar to a dark/light mode toggle), which could modify inline as well?

**Comment by ghost** [comment]
Would be great to see more about handling and optimizing previews with complex GROQ queries. I was able to remove the old iFrame plugin and API preview code by using components and hitting the groq queries directly. I'm still having trouble with preview load speeds and no hot reloading on editor updates.

**Reply by kmelve** [reply]
For sure @jasonbarone! Improving the developer/editor experience for implementing previews is also something we're looking at. 

Btw. did you see this [new documentation article on performant GROQ](https://www.sanity.io/docs/high-performance-groq)?

**Comment by matt416** [comment]
I'm really interested in input component customization. Would love to see docs and example code

**Comment by vincentjflorio** [comment]
>What do you expect to be covered in the Sanity Studio docs?

The original docs are very comprehensive, but there are plenty of tasks that people do all the live-long day that could get extra attention, like the image URL builder. I think they're covered well, even but just to _really_ lay it on thick, perhaps with a different presentation each time, to make sure as many people with as many different learning styles truly get it.

Maybe even some wow-factor-but-genuinely-practical-because-people-actually-do-it examples, like async operations that fill in a value in a field. During the preview presentation I loved seeing the USDA autocomplete. I could see so many businesses using something like that even if it were dogfooding their own API.

If developers are experienced with a language, or at coding, that doesn't mean they might not want to see an application (ha) of those skills in context.

If they're starting out, they might be more engaged if it looks like what they're learning after those basics is something they could get a lot of utility out of.

Also the API reference, it was said, is self-generating but unless I'm missing it, when the time comes it'd be cool to have a live-filtering or a search. There is a *lot* of information and if you stare at the names for too long it sort of bleeds together. Especially with hooks. So if I go in there knowing what I want to refer to, it'd be great to limit the list to the six things that look like that.

>Is there anything that you think works well?

Even though there are just a couple pages of the new version, I like the idea that it might build cumulatively. In other words, yes, you can know about each topic at a glance, but if you're already holding my hand by teaching me about schemas with pets that have names, perhaps by the time we roll around to list previews, we're still building on what we made before so we can see how it all ties together.

**Comment by jkananen** [comment]
What's the best way to get notified of all beta docs updates? I tried to look into your repos, but couldn't find anything that I could subscribe to for this.

**Reply by kmelve** [reply]
We don't have any mechanism for that… yet. Maybe we could make a small changelog thingy that relays a list of the last updated articles!

**Reply by jkananen** [reply]
It would be much appreciated. Going to the beta docs every day to look for updates is not very efficient 😄

If you do it with a changelog listing, links to the updated pages would be great!

**Reply by kmelve** [reply]
We shipped [the changelog here](https://beta.sanity.io/docs/changelog).

**Comment by vincentjflorio** [comment]
I just thought of another "coverage" thing -- I am reading Carrie and Mike's "Connected Content" book and it's quietly tangible that even though structured content might be a strong alternative for some companies and use cases to the way things have been done up until recently, it's all still "living" information; about midway through one of the interviewees is talking about how well some of their decisions have held up, but in the Slack channels it's a reasonably common occurrence that people  are talking about finding-and-replacing fields, switching types, formats and such.

If that's going to be people's real-life experience, it might be good to give separate attention to move-making like that -- not because it's v3-specific so much as, if you're already going to deep-dive and massage these things, might as well take advantage of the diving sessions.

The mechanisms are in place to do it, but touching anything that might be altered or affective even if you have backups is a scary thing no matter where you are in your development expertise. So something really clear and over about how to handle things we might run into like that. Big important "steps" in the live of a ...er...Sanitarian. To keep the ships from sinking so that all these really thoughtful decisions before coding and all these snazzy conveniences during coding continue to stay afloat and work ship-shape. (I really can't help myself)

**Reply by kmelve** [reply]
I think you're talking about evolving content models and how to migrate data when structures change. I agree that it would be useful to cover some common strategies for this in the docs!

**Comment by vincentjflorio** [comment]
I had this thought, but maybe it's silly. I can't tell you how many times I read an example of code somewhere and I have an immediate thought related to it.

For example, if I am on a remote database's documentation, and it says "We have the simplest auth system in the world," and they show really basic examples that all involve five fields, like ID, first name, last name, etc. but then the final example is, "and here's how you'd make a function that triggers the addition of a new user to the table when they've completed signup" and *that* example only pushes the ID in...

...I am now super curious how to add in all the non-ID stuff if the format is completely new to me (in other words, if it's not as simple as just adding more things to the list, copying pasting, what not...basic developer inference skills.

It might be cool if inline in the documentation there was a button to ask a question, as if to say "I want to know specifically about *this*" and fire off those questions sort of pre-categorized for context. Whether that goes here, in Slack, to your own internal tool, whatnot.

So much of asking questions nowadays is hesitancy, because of the "getting on the same page," the volleying back and forth. In real life you'd pull someone aside and say, "What's that bit mean." "Oh it means this," so having the direct connection to the information established right out of the gate, with context, feels like it maybe eliminates some of those barriers.

**Reply by kmelve** [reply]
Interesting idea @vincentjflorio! Most of our effort towards GA will be on just getting the v3 APIs documented. But we're definitively interested in adding more helpful features to the doc site as well.

**Comment by ottob** [comment]
Something that took me a while today to figure out was how to get the dataset (production/development) from an environment variable.
I found the documentation here: https://beta.sanity.io/docs/platform/studio/v2-to-v3/environment-variables#bundling-your-own-variables-in-the-studio-application and that worked well for sanity.config.ts but using `import.meta.env.SANITY_STUDIO_DATASET` did not work in my sanity.cli.ts, there I had to use `process.env.SANITY_STUDIO_DATASET` instead.
This makes sense since vite is not running in the cli. But I suspect more users might run into this issue.

**Reply by ottob** [reply]
Upon further testing I can't get this to work properly. I have a `.env.development` file containing `SANITY_STUDIO_DATASET=development` and a `.env.production` containing `SANITY_STUDIO_DATASET=production`.

I have removed my `dataset` property from `sanity.cli.ts` and it is only specified in `sanity.config.ts` as:
```ts
dataset: import.meta.env.SANITY_STUDIO_DATASET,
```

But when I run `sanity deploy` I get the _development_ dataset on the _production_ studio site. This is not what I expected having read: https://www.sanity.io/docs/studio-environment-variables#f5e9e3158896

When I run `sanity deploy` it says:
```
Including the following environment variables as part of the JavaScript bundle:
- SANITY_STUDIO_DATASET
```
It would be helpful if it printed the value also.

Im running sanity v3.0.0-dev-preview.15

**Reply by ottob** [reply]
In the [v2 docs](https://www.sanity.io/docs/studio-environment-variables) `SANITY_STUDIO_API_DATASET` is used but the [v3 docs](https://beta.sanity.io/docs/platform/studio/v2-to-v3/environment-variables) only refers to `SANITY_STUDIO_DATASET`. Is the v2 one deprecated?

**Reply by duannx** [reply]
I'm surprised that this issue has existed for months with any v3 project but still get no response

**Comment by intensr** [comment]
Maybe add an example here https://beta.sanity.io/docs/platform/studio/v2-to-v3/custom-input for a custom block content / portable preview component, like how to customize it in v3 style

**Comment by trevorpfiz** [comment]
Is the blockEditor field mentioned [here](https://www.sanity.io/docs/customization) still a thing?

[I see that there is a discussion here](https://github.com/sanity-io/sanity/discussions/3902)
